restapi/src/main/java/io/stargate/web/docsapi/service/JsonDocumentShredder.java

```diff
@@ -0,0 +1,174 @@
+          e);
+    }
+  }
+
+  public List<JsonShreddedRow> shred(
+      JsonNode node, String documentId, List<String> subDocumentPath, boolean numericBooleans) {
+    Supplier<ImmutableJsonShreddedRow.Builder> rowBuilder =
+        () ->
+            ImmutableJsonShreddedRow.builder()
+                .key(documentId)
+                .maxDepth(config.getMaxDepth())
+                .addAllPath(subDocumentPath); // TODO should we also convert these parts with escape
+
+    List<JsonShreddedRow> result = new ArrayList<>();
+    processNode(node, rowBuilder, numericBooleans, result);
+    return result;
+  }
+
+  private void processNode(
+      JsonNode node,
+      Supplier<ImmutableJsonShreddedRow.Builder> rowBuilder,
+      boolean numericBooleans,
+      List<JsonShreddedRow> result) {
+    if (node.isArray()) {
+      processArrayNode(node, rowBuilder, numericBooleans, result);
+    } else if (node.isObject()) {
+      processObjectNode(node, rowBuilder, numericBooleans, result);
+    } else {
+      processValueNode(node, rowBuilder, numericBooleans, result);
+    }
+  }
+
+  private void processArrayNode(
+      JsonNode node,
+      Supplier<ImmutableJsonShreddedRow.Builder> rowBuilder,
+      boolean numericBooleans,
+      List<JsonShreddedRow> result) {
+    // empty array, simply create a reference to empty node and return
+    if (node.isEmpty()) {
+      ImmutableJsonShreddedRow row =
+          rowBuilder.get().stringValue(DocsApiConstants.EMPTY_ARRAY_MARKER).build();
+      result.add(row);
+      return;
+    }
+
+    // otherwise, iterate all nodes
+    int idx = 0;
+    for (JsonNode inner : node) {
+      // make sure we didn't exceed the maximum array length
+      if (idx >= config.getMaxArrayLength()) {
+        throw new ErrorCodeRuntimeException(ErrorCode.DOCS_API_GENERAL_ARRAY_LENGTH_EXCEEDED);
+      }
+
+      // convert the array index into path
+      // then create new next row builder
+      String arrayPath = "[" + DocsApiUtils.leftPadTo6(String.valueOf(idx)) + "]";
+      Supplier<ImmutableJsonShreddedRow.Builder> nextRowBuilder =
+          () -> rowBuilder.get().addPath(arrayPath);
+
+      // process inner node and increase the index
+      processNode(inner, nextRowBuilder, numericBooleans, result);
+      idx++;
+    }
+  }
+
+  private void processObjectNode(
+      JsonNode node,
+      Supplier<ImmutableJsonShreddedRow.Builder> rowBuilder,
+      boolean numericBooleans,
+      List<JsonShreddedRow> result) {
+    // empty object, simply create a reference to empty node and return
+    if (node.isEmpty()) {
+      ImmutableJsonShreddedRow row =
+          rowBuilder.get().stringValue(DocsApiConstants.EMPTY_OBJECT_MARKER).build();
+      result.add(row);
+      return;
+    }
+
+    node.fields()
+        .forEachRemaining(
+            field -> {
+              // TODO is this still valid
+              //  ErrorCode.DOCS_API_GENERAL_INVALID_FIELD_NAME
```

Comment:
@EricBorczuk what about this one? Is the field name restricted at the moment? I am seeing this in the current shredding:

```
                if (fieldName != null && DocsApiUtils.containsIllegalSequences(fieldName)) {
                  String msg =
                      String.format(
                          "Array paths contained in square brackets, periods, single quotes, and backslash are not allowed in field names, invalid field %s",
                          fieldName);
                  throw new ErrorCodeRuntimeException(
                      ErrorCode.DOCS_API_GENERAL_INVALID_FIELD_NAME, msg);
                }
```

Reply:
Oh yes - sorry I didn't see that this was removed. Putting these characters in without explicitly escaping them with `\` should be disallowed

Reply:
OK, hope I got this right, can you maybe quickly check my tests for these?